common/database.cpp

```diff
@@ -412,23 +412,40 @@ bool Database::DeleteCharacter(char *character_name) {
 
 		QueryDatabase(query);
 
-		return true;
```

Comment:
Instead of removing the return here, just inline the bots preprocessor in the existing `if (RuleB(Character, SoftDeletes)) {`